core/src/main/scala/org/bykn/bosatsu/TypedExprNormalization.scala

```diff
@@ -119,6 +119,16 @@ object TypedExprNormalization {
   private def setType[A](expr: TypedExpr[A], tpe: Type): TypedExpr[A] =
     if (!tpe.sameAs(expr.getType)) Annotation(expr, tpe) else expr
 
+  private def appLambda[A](f1: AnnotatedLambda[A], args: NonEmptyList[TypedExpr[A]], tpe: Type, tag: A): TypedExpr[A] = {
+    val AnnotatedLambda(lamArgs, expr, _) = f1
+    // (y -> z)(x) = let y = x in z
```

Comment:
This law seems suspect. What if the args have free variables that intersect with the lambda args? It isn't this simple.